tests/framework/e2e/etcd_process.go

```diff
@@ -179,3 +183,14 @@ func (ep *EtcdServerProcess) Logs() LogsExpect {
 	}
 	return ep.proc
 }
+
+func ExpectLog(t *testing.T, ep EtcdProcess, expectLog string) {
+	t.Helper()
+	var err error
+	testutils.ExecuteWithTimeout(t, 30*time.Second, func() {
+		_, err = ep.Logs().ExpectWithContext(context.Background(), expectLog)
+	})
```

Comment:
It seems that there is no need to use `testutils.ExecuteWithTimeout` at all, you can use `ExpectWithContext` directly. You just need to create a context with Timeout, and pass it to `ExpectWithContext`

Reply:
Done